src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -326,6 +326,60 @@ export const handleGetPublicForm: ControllerHandler<
   }
 }
 
+export const handleGetPublicFormSampleSubmission: ControllerHandler<
+  { formId: string },
+  Record<string, any> | ErrorDto | PrivateFormErrorDto
+> = async (req, res) => {
+  const { formId } = req.params
+  const logMeta = {
+    action: 'handleGetPublicFormSampleSubmission',
+    ...createReqMeta(req),
+    formId,
+  }
+
+  const formResult = await getFormIfPublic(formId)
+  // Early return if form is not public or any error occurred.
+  if (formResult.isErr()) {
+    const { error } = formResult
+    // NOTE: Only log on possible database errors.
+    // This is because the other kinds of errors are expected errors and are not truly exceptional
+    if (isMongoError(error)) {
+      logger.error({
+        message: 'Error retrieving public form',
+        meta: logMeta,
+        error,
+      })
+    }
+    const { errorMessage, statusCode } = mapRouteError(error)
+
+    // Specialized error response for PrivateFormError.
+    // This is to maintain backwards compatibility with the middleware implementation
+    if (error instanceof PrivateFormError) {
+      return res.status(statusCode).json({
+        message: error.message,
+        // Flag to prevent default 404 subtext ("please check link") from
+        // showing.
+        isPageFound: true,
+        formTitle: error.formTitle,
+      })
+    }
+
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+  const form = formResult.value
+
+  const publicForm = form.getPublicView() as PublicFormDto
+
+  const sampleData: Record<string, any> = {}
+  const formFields = publicForm.form_fields
+  if (!formFields) {
+    throw new Error('unable to get form fields')
+  }
+  for (const field of formFields) {
+    FormService.createSampleSubmissionData(sampleData, field)
+  }
```

Comment:
/nit: For code readability/cleanliness, it's better to write this as a [pure function](https://en.wikipedia.org/wiki/Pure_function) that return a sample response for a given field `createSampleSubmissionAnswer(field)`, and do the mutation of the result object here (as opposed to passing the result object as first arg and mutating it with every function call).

```suggestion
  const sampleData: Record<string, any> = formFields.reduce((acc, field) => {
    const answer = FormService.createSampleSubmissionAnswer(field)
    if (!answer) return acc
    acc[field._id] = answer
    return acc
  }, {})
```

If you'd like to reduce the code in the controller, create an additional method `FormService.createSampleSubmissionResponses()`, such that the route handler can just invoke it:

```typescript
  const publicForm = form.getPublicView() as PublicFormDto
  const formFields = publicForm.form_fields

  if (!formFields) {
    throw new Error('unable to get form fields')
  }
  
  const responses = FormService.createSampleSubmissionResponses(publicForm.form_fields)
  
  return res.json({ responses })
```

And inside `FormService.createSampleSubmissionResponses`, you can call `FormService.createSampleSubmissionAnswer`.

Reply:
Thanks Tim! Created an additional method in `FormService`

[3f2dcc1](https://github.com/opengovsg/FormSG/pull/6325/commits/3f2dcc19ac6a66253f05f172c83cb3dd2d70a458)